### PR TITLE
Running a Multi-container (Spring Boot and PostgreSQL) Application with Docker Compose

### LINKED ISSUE #2737: Getting started with Docker Compose using Spring Boot App and PostgreSQL containers
### Proposed title of article
Getting started with Docker Compose using Spring Boot App and PostgreSQL containers.

### Introduction paragraph (2-3 paragraphs):

Container technology has streamlined how we build, test and deploy software from our local environment to on-premise data centres and the cloud. With the benefit of building applications with container technology, as we build multi-container applications, manually starting and stopping each container becomes tedious. 

Docker-compose is a tool used in defining and running multi-container Docker applications. We use Docker-compose to replace the run commands repeatedly run on each container in a multi-container application. Using Docker compose, we can simplify the running of multi-container applications to as little as two commands; `docker-compose up` and `docker-compose down`.

In this tutorial, we’ll learn how Docker-compose makes it easier to configure and run multiple Docker containers in a local environment. We will also create Spring Boot application and PostgreSQL database containers.

### Key takeaways: 

1. Understand the basics of Docker Compose and why developers should use it in their workflow.
2. Learn how to create a PostgreSQL Docker image.
3. Learn how to define a docker-compose.yml file. 

### References: 

- https://www.youtube.com/watch?v=QeQ2MH5f_BE 
- https://github.com/Kikiodazie/Multi-User-TodoList-Api
- https://www.youtube.com/watch?v=HUpIoF_conA

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Kikiodazie**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

this fixes #2737

### CODE REVIEW

**Comment by Kikiodazie** [comment]
This fixes issue #2737. I changed the title from "Getting started with Docker Compose using Spring Boot App and PostgreSQL containers" to "Running a Multi-container (Spring Boot and PostgreSQL) Application with Docker Compose"

**Comment by Kikiodazie** [comment]
@paulodhiambo hey, good day. Here is the article, I am looking forward to your review.

**Comment by ahmadmardeni1** [comment]
Hey @Kikiodazie 👋
Paulo has 5 articles to review, and you can't choose a peer reviewer, someone else will review this PR. 
Thanks for understanding!

**Comment by Kikiodazie** [comment]
> Hey @Kikiodazie 👋
> Paulo has 5 articles to review, and you can't choose a peer reviewer, someone else will review this PR.
> Thanks for understanding!

Ohhh sorry. No problem. Thanks for the clarification. Looking forward to the review.

**Comment by Kikiodazie** [comment]
Hey @ahmadmardeni1 good day. I am just checking up on this.

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. :rocket:
Thank you. @Kikiodazie

**Comment by Kikiodazie** [comment]
Thank you @ahmadmardeni1.

Thank you @Bayler Looking forward to the review.

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
+Container technology has streamlined how we build, test and deploy software from our local environment to on-premise data centers and the cloud. With the benefit of building applications with container technology, manually starting and stopping each container becomes tedious as you build multi-container applications. 
+
+Docker Compose is a tool used in defining and running multi-container Docker applications. We use Docker Compose to replace the `run` commands that is repeatedly run on each container in a multi-container application. Using Docker Compose, we can simplify the running of multi-container applications to as little as two commands; `docker-compose up` and `docker-compose down`.
+
+### Goal
+
+In this tutorial, you’ll learn how Docker Compose makes it easier to configure and run multiple Docker containers in a local environment. For demo purposes, this tutorial will use a Spring Boot and a PostgreSQL container.
```
Comment (Bayler) [inline content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md:7]: Please add external links to both Spring Boot and PostgreSQL websites.

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+### Prerequisites
+
+- Basic knowledge of Spring Boot, Git, and the terminal.
+- Basic understanding of Docker and Docker Compose installed - see guide [here](https://docs.docker.com/compose/install/).
+- A Java IDE - In this tutorial, we will use [IntelliJ Idea](https://www.jetbrains.com/idea/), but you can use any IDE of your choice.
+- Postman - download it [here](https://www.postman.com/downloads/)
```
Comment (Bayler) [inline content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md:14]: Please add a single sentence explaining what Postman is before the "download it here" section. This is beneficial to a reader

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+In the `application.properties` file add:
+
+```bash
+    spring.datasource.driverClassName=org.postgresql.Driver
+```
+
+The above configuration tells Spring Boot that the data source will be of PostgreSQL using PostgreSQL’s specific driver.
+
+Normally when setting up a spring boot project to connect to a data source, we use similar to in the `application.properties` file:
+
```
Comment (Bayler) [inline content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md:44]: This sentence is not clear...."we use similar to in the...." is there something missing?

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md (hunk 1 above)
Comment (Kikiodazie) [reply content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md:7]: Ok then, will do that right away.

Like this below right?

[Spring Boot](https://spring.io/projects/spring-boot) and [PostgreSQL](https://www.postgresql.org/)

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md (hunk 2 above)
Comment (Kikiodazie) [reply content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md:14]: Noted.

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md (hunk 1 above)
Comment (Bayler) [reply content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md:7]: Yes. That's right

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md (hunk 3 above)
Comment (Kikiodazie) [reply content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md:44]: Ohhh. will look into it right away.

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md (hunk 1 above)
Comment (Kikiodazie) [reply content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md]: Done

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md (hunk 2 above)
Comment (Kikiodazie) [reply content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md]: Done. Do check it out. Made the right edit in the last commit.

File: content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md (hunk 3 above)
Comment (Kikiodazie) [reply content/articles/running-a-multi-container-springboot-postgresql-application-with-docker-compose/index.md]: Done. Do check it out.

**Comment by Bayler** [comment]
@Kikiodazie Great article! I love the way you've explained the code in detail. Your article also has very minimal corrections. Keep up the good work. This article is now ready for final review.

**Comment by Kikiodazie** [comment]
> @Kikiodazie Great article! I love the way you've explained the code in detail. Your article also has very minimal corrections. Keep up the good work. This article is now ready for final review.

Thanks, @Bayler. I appreciate your feedback. 


And thank you for reviewing my article.

**Comment by LinusMuema** [comment]
Hi @Kikiodazie 
Thanks for this amazing article and keep up the amazing job.
Check my commit for more changes.

cc: @Bayler

**Comment by Kikiodazie** [comment]
> Hi @Kikiodazie
> Thanks for this amazing article and keep up the amazing job.
> Check my commit for more changes.
> 
> cc: @Bayler

Thank you @LinusMuema. I appreciate this.

**Comment by Kikiodazie** [comment]
@LinusMuema Sorry I check the article and it seems the uploaded images aren't showing.

**Comment by LinusMuema** [comment]
Let me have a look

**Pull request merged**